twill-yarn/src/test/java/org/apache/twill/yarn/ProvisionTimeoutTestRun.java

```diff
@@ -79,12 +83,22 @@ public void initialize(EventHandlerContext context) {
 
     @Override
     public TimeoutAction launchTimeout(Iterable<TimeoutEvent> timeoutEvents) {
+      for (TimeoutEvent event : timeoutEvents) {
+        LOG.info("Requested {} containers for runnable {}, only got {} after {} ms.",
+                 event.getExpectedInstances(), event.getRunnableName(),
+                 event.getActualInstances(), System.currentTimeMillis() - event.getRequestTime());
+      }
       if (abort) {
         return TimeoutAction.abort();
       } else {
         return TimeoutAction.recheck(10, TimeUnit.SECONDS);
       }
     }
+
+    @Override
+    public void aborted(String twillAppName, RunId runId) {
+      LOG.info(String.format("Aborted %s with runId %s", twillAppName, runId.getId()));
```

Comment:
Besides logging, can you add/modify the test to actually validate the new lifecycle methods in the EventHandler?